lldb/tools/lldb-dap/DAP.cpp

```diff
@@ -36,9 +36,7 @@ DAP::DAP()
           {{"cpp_catch", "C++ Catch", lldb::eLanguageTypeC_plus_plus},
            {"cpp_throw", "C++ Throw", lldb::eLanguageTypeC_plus_plus},
            {"objc_catch", "Objective-C Catch", lldb::eLanguageTypeObjC},
-           {"objc_throw", "Objective-C Throw", lldb::eLanguageTypeObjC},
-           {"swift_catch", "Swift Catch", lldb::eLanguageTypeSwift},
-           {"swift_throw", "Swift Throw", lldb::eLanguageTypeSwift}}),
+           {"objc_throw", "Objective-C Throw", lldb::eLanguageTypeObjC}}),
```

Comment:
See my comments for this patch in the review. It would be great to determine if a language is supported by LLDB dynamically and only add support for languages that the current LLDB supports. So we would probably need to default construct the `exception_breakpoints()` and then inside this constructor or before we try to use `exception_breakpoints`, we would need to populate it by doing something like:
```
if (g_dap.debugger.SupportsLanguage(lldb::eLanguageTypeC_plus_plus)) {
  exception_breakpoints.emplace_back({"cpp_catch", "C++ Catch", lldb::eLanguageTypeC_plus_plus})
  exception_breakpoints.emplace_back("cpp_throw", "C++ Throw", lldb::eLanguageTypeC_plus_plus})
}
if (g_dap.debugger.SupportsLanguage(lldb::eLanguageTypeObjC)) {
  exception_breakpoints.emplace_back({"objc_catch", "Objective-C Catch", lldb::eLanguageTypeObjC})
  exception_breakpoints.emplace_back("objc_throw", "Objective-C Throw", lldb::eLanguageTypeObjC})
}
if (g_dap.debugger.SupportsLanguage(lldb:: eLanguageTypeSwift)) {
  exception_breakpoints.emplace_back({"swift_catch", "Swift Catch", lldb:: eLanguageTypeSwift})
  exception_breakpoints.emplace_back("swift_throw", "Swift Throw", lldb:: eLanguageTypeSwift})
}
```

Reply:
Thanks for the detailed suggestions! Makes sense!

For ObjC, just to clarify, the intention of hiding the filters was NOT because the debugger didn't support it - the intention was to avoid clustering/confusing users - that is, if they're not debugging ObjC, there's no reason to show the objc filters. (And we infer this by checking if they're debugging on a mac - it's theoretically possible they're cross debugging objc from a non-mac but in practice I don't think it's likely )
Does this seem reasonable?

Reply:
The ObjC Language runtime is able to detect whether ObjC is loaded into the program.  It runs that detector on each library load, so it keeps an accurate notion of this.  Ditto for the swift language runtime.  It would be better to consult that than guess based on platform.  There may not currently be a way to ask that from the SB API's but we can certainly make one...

Reply:
> The ObjC Language runtime is able to detect whether ObjC is loaded into the program. It runs that detector on each library load, so it keeps an accurate notion of this. Ditto for the swift language runtime. It would be better to consult that than guess based on platform.

This code runs before the runtimes (objc/swift) so it probably won't be able to query that, though?

Reply:
P.S: it was pointed out to me that changed capabilities can be pushed as an event. So I guess we could init the list of filters based on whether the languages are supported. Then when the runtime(s) become available, we can update the list after querying for which language is being debugged. WDYT?

Reply:
ObjC tends to only use exceptions for error reporting, not for message passing, so a lot of ObjC developers leave this breakpoint on all the time.  Having to wait till the first load event that brings in ObjC would make that tedious.

Reply:
Btw, it's possible to register typesystems/languages at initialization via runtime plugins. I think it would be ideal if the breakpoint list could get updated/refreshed after the target has been loaded (maybe at the first stop point)